scheduler/pkg/envoy/processor/incremental.go

```diff
@@ -640,8 +647,8 @@ func (p *IncrementalProcessor) modelUpdate(modelName string) error {
 func (p *IncrementalProcessor) callVersionCleanupIfNeeded(modelName string) {
 	logger := p.logger.WithField("func", "callVersionCleanupIfNeeded")
 	if routes, ok := p.xdsCache.Routes[modelName]; ok {
-		activeRoutes := len(routes.Clusters)
-		if activeRoutes == 1 && p.versionCleaner != nil {
```

Comment:
We do not anymore need to do this check. It was also buggy because this didnt capture edge cases where the routes also point to a candidate model in the case of an experiment with a default model route.